sklearn/linear_model/_glm/glm.py

```diff
@@ -267,6 +280,31 @@ def fit(self, X, y, sample_weight=None):
             )
             self.n_iter_ = _check_optimize_result("lbfgs", opt_res)
             coef = opt_res.x
+        elif self.solver == "newton-cholesky":
+            sol = NewtonCholeskySolver(
+                coef=coef,
+                linear_loss=linear_loss,
+                l2_reg_strength=l2_reg_strength,
+                tol=self.tol,
+                max_iter=self.max_iter,
+                n_threads=n_threads,
+                verbose=self.verbose,
+            )
+            coef = sol.solve(X, y, sample_weight)
+            self.n_iter_ = sol.iteration
+        elif issubclass(self.solver, NewtonSolver):
+            sol = self.solver(
+                coef=coef,
+                linear_loss=linear_loss,
+                l2_reg_strength=l2_reg_strength,
+                tol=self.tol,
+                max_iter=self.max_iter,
+                n_threads=n_threads,
+            )
+            coef = sol.solve(X, y, sample_weight)
+            self.n_iter_ = sol.iteration
+        else:
+            raise TypeError(f"Invalid solver={self.solver}.")
```

Comment:
We should need this because the solver should be validated in the `_parameter_constraints`.

Reply:
Ups wrong PR but I still think that we should not need this.

Reply:
I'll change it to `ValueError` which seems our standard here. But it can happen, e.g.
```python
class MyClass:
    pass

PoissonRegressor(solver=MyClass).fit([[1]], [2])
```

Reply:
Oh, I see. So this is the reason that we support `Hidden(type)`.
Actually, the clean way would be to define the interface that we expect a solver class using `HasMethod(["setup", ...])`.

Maybe we don't have yet a good idea of the interface and in this case, I understand that this is safer to keep the `else` and raise the error ourselves because `type` is too much permissive.